### PR TITLE
Enable more rules for detekt code base

### PR DESCRIPTION
**Pull request by marschwar**
There are quite a few rule that are deactivated by default and therefor are not checked when running checking the detekt code base.  I think it is very helpful to at least run those that are not controversial.

Looking back I should have probably created multiple PR for this. Sorry about that.

There are still some rules I would like to enable but that would blow up the change set even more. So there will be more PR unless you think there is no need for that.

### CODE REVIEW

File: detekt-metrics/src/main/kotlin/io/github/detekt/metrics/ComplexityMetric.kt [hunk 1]
```diff
@@ -7,6 +7,7 @@ import io.github.detekt.metrics.processors.logicalLinesKey
 import io.github.detekt.metrics.processors.sourceLinesKey
 import io.gitlab.arturbosch.detekt.api.Detektion
 
+@Suppress("UseDataClass")
```
Comment (marschwar) [inline detekt-metrics/src/main/kotlin/io/github/detekt/metrics/ComplexityMetric.kt:10]: This feels like a false positive to me. Should we create an issue for that?

File: detekt-rules-documentation/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/LicenceHeaderLoaderExtension.kt [hunk 2]
```diff
@@ -57,6 +57,7 @@ class LicenceHeaderLoaderExtension : FileProcessListener {
                 """.trimIndent()
             }
 
+            @Suppress("UnnecessaryLet")
             return Files.newBufferedReader(templateFile)
                 .use(BufferedReader::readText)
                 .let(StringUtilRt::convertLineSeparators)
```
Comment (marschwar) [inline detekt-rules-documentation/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/LicenceHeaderLoaderExtension.kt:63]: This feels like a false positive to me. Should we create an issue for that?

File: detekt-metrics/src/main/kotlin/io/github/detekt/metrics/ComplexityMetric.kt [hunk 3]
```diff
@@ -7,6 +7,7 @@ import io.github.detekt.metrics.processors.logicalLinesKey
...
 class ComplexityMetric(detektion: Detektion) {
 
     val mcc = detektion.getData(complexityKey)
```
Comment (marschwar) [inline detekt-metrics/src/main/kotlin/io/github/detekt/metrics/ComplexityMetric.kt:13]: This feels like a false positive to me. Should we create an issue for that?

File: detekt-rules-documentation/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/LicenceHeaderLoaderExtension.kt (hunk 2 above)
Comment (schalkms) [reply detekt-rules-documentation/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/LicenceHeaderLoaderExtension.kt:63]: Yes, please

File: detekt-metrics/src/main/kotlin/io/github/detekt/metrics/ComplexityMetric.kt (hunk 3 above)
Comment (schalkms) [reply detekt-metrics/src/main/kotlin/io/github/detekt/metrics/ComplexityMetric.kt:13]: This is also a false-positive.

Comment (marschwar) [reply detekt-metrics/src/main/kotlin/io/github/detekt/metrics/ComplexityMetric.kt]: On second thought I believe this should be a data class with a factory method instead.

**Comment by marschwar** [comment]
This change set is way to big. I will close this and start from scratch with multiple PR.

**Pull request closed without merging**